src/pages/MerchDetail/MerchDetail.tsx

```diff
@@ -73,25 +75,36 @@ export const MerchDetail: React.FC = () => {
     const value = parseInt(target.value, 10);
     if (value <= 0) {
       setQuantity(1);
-    } else if (value > 99) {
-      setQuantity(99);
+    } else if (value > maxQuantity) { 
+      setQuantity(maxQuantity);
     } else {
       setQuantity(value);
     }
   };
 
+  const updateMaxQuantity = (colorway: string, size: string) => {
+    if (product) {
+      const stockQty = getQtyInStock(product, colorway, size);
+      const cartQty = getQtyInCart(cartState.items, product.id, colorway, size);
+      const max = (stockQty > cartQty) ? stockQty - cartQty : 0;
+      setMaxQuantity(max);
+    }
+  }
+
   const handleAddToCart = () => {
     setIsDisabled(true);
     const payload: CartAction = {
       type: CartActionType.ADD_ITEM,
       payload: {
         productId,
         quantity,
-        size: selectedSize ?? product?.sizes?.[0] ?? "",
-        colorway: selectedColorway ?? ""
+        colorway: selectedColorway ?? "", // TODO ? 
+        size: selectedSize ?? "", // TODO ?
```

Comment:
@chanbakjsd not sure what the default colorway/size should be
(or if it even matters, since I disabled the "add to cart" and "buy now" buttons until both are selected)

Reply:
You should validate that `selectedColorway` and `selectedSize` is not null, that should allow you to just assign it directly. If it is null, you might want to just return without doing anything.